viewer/src/migration/Cognite3DViewer.ts

```diff
@@ -0,0 +1,145 @@
+/*!
+ * Copyright 2020 Cognite AS
+ */
+
+import * as THREE from 'three';
+import ComboControls from '@cognite/three-combo-controls';
+
+import { Cognite3DModel } from './Cognite3DModel';
+import { Cognite3DViewerOptions, AddModelOptions } from './types';
+import { NotSupportedInMigrationWrapperError } from './NotSupportedInMigrationWrapperError';
+import { Intersection } from './intersection';
+
+export class Cognite3DViewer {
+  private get canvas(): HTMLCanvasElement {
+    return this.renderer.domElement;
+  }
+  static isBrowserSupported(): boolean {
+    return true;
+  }
+
+  readonly domElement: HTMLElement;
+  private readonly renderer: THREE.WebGLRenderer;
+  private readonly camera: THREE.PerspectiveCamera;
+  private readonly scene: THREE.Scene;
+  private readonly controls: ComboControls;
+
+  private readonly models: Cognite3DModel[] = [];
+  private readonly additionalObjects: THREE.Object3D[] = [];
+
+  constructor(options?: Cognite3DViewerOptions) {
+    if (options && !options.enableCache) {
+      throw new NotSupportedInMigrationWrapperError('Cache is not supported');
+    }
+    if (options && !options.logMetrics) {
+      throw new NotSupportedInMigrationWrapperError('LogMetris is not supported');
+    }
+
+    this.renderer = new THREE.WebGLRenderer();
+    this.canvas.style.width = '640px';
+    this.canvas.style.height = '480px';
+    this.canvas.style.minWidth = '100%';
+    this.canvas.style.minHeight = '100%';
+    this.canvas.style.maxWidth = '100%';
+    this.canvas.style.maxHeight = '100%';
+    this.domElement = createCanvasWrapper();
+    this.domElement.appendChild(this.canvas);
+
+    this.camera = new THREE.PerspectiveCamera(60, undefined, 0.1, 10000);
+    this.camera.position.x = 30;
+    this.camera.position.y = 10;
+    this.camera.position.z = 50;
+    this.camera.lookAt(new THREE.Vector3());
+
+    this.scene = new THREE.Scene();
+    this.controls = new ComboControls(this.camera, this.canvas);
+    this.controls.dollyFactor = 0.992;
+    this.controls.enabled = false;
+  }
+
+  dispose(): void {
+    throw new NotSupportedInMigrationWrapperError();
+  }
+
+  onClick(callback: (event: PointerEvent) => void): void {
```

Comment:
Why is this not wrapped the same way as before, with `on('click'`) as the pattern?

Reply:
I will try to change this, but I was not able to make TypeScript accept on('click') and on('cameraChange') etc with correct event type. If you manage to fix it your my hero for the day.